packages/babel-plugin-proposal-object-rest-spread/src/index.js

```diff
@@ -240,6 +240,39 @@ export default declare((api, opts) => {
     }
   }
 
+  function doesAnyChildNodeHaveMoreThanOneProperty(node) {
+    if (node.argument) return true;
```

Comment:
If `node.argument` presents, it is a `RestElement` and we need to recurse to `node.argument`, otherwise we don't know if it contains more than one bindings:

```js
const [...{...a}] = []
```
When we see the first `...` after `[`, we should look into the AST and recurse to `{...a}`.

Reply:
Added the test cases below and updated as per other comments
```js
const [...[ ...y17 ]] = z();
const [...{ a19, ...y19 }] = z();
```